### PR TITLE
Add documentation for CaaSP deployment on OpenStack

### PR DESCRIPTION
**Pull request by greygoo**

### CODE REVIEW

File: Notes-CaaSP.md [hunk 1]
```diff
@@ -132,11 +132,177 @@ kubectl create secret generic ceph-secret-admin
...
-Use the free space, create a partition on it, format it with btrfs and mount it as
-`/var/lib/docker` before you start importing any containers. If you resize the root filesystem to the full disk, you have very big root filesystem and still no space to store the containers without a big performance penalty.
+* Provide enough disk space for images
+
+  The default is to provide 40 GB of disk space to the CaaSP nodes. This is not sufficient for CAP. So use a machine flavor with a bigger disk and let it use the additional free space for storing images (Commands on how to resize the CaaSP node root fs are provided in the instructions below).
+
+
+### Initial preparations
+
+The following steps only have to be done once before the initial CaaSP deployment. For following deplyments of CaaSP this has no to be redone but already created elements can be reused. The deployment of CaaSP on OpenStack is done using existing terraform rules with some additional steps required for running CAP on the CaaSP deployment.
```
Comment (cornelius) [inline Notes-CaaSP.md]: Typo "deplyments"

File: Notes-CaaSP.md [hunk 2]
```diff
@@ -132,11 +132,177 @@ kubectl create secret generic ceph-secret-admin
...
+
+  ```
+  git clone git@github.com:kubic-project/automation.git
+  cd automation/caasp-openstack-terraform
+  ```
+
+* Upload the [CaaSPv2 image](http://dist.suse.de/install/SUSE-CaaSP-2.0-GM/SUSE-CaaS-Platform-2.0-for-OpenStack-Cloud.x86_64-2.0.0-GM.qcow2)
+
+  ```
+  wget http://dist.suse.de/install/SUSE-CaaSP-2.0-GM/SUSE-CaaS-Platform-2.0-for-OpenStack-Cloud.x86_64-2.0.0-GM.qcow2
```
Comment (cornelius) [inline Notes-CaaSP.md]: This is an internal URL. We should use an external one.

File: Notes-CaaSP.md [hunk 3]
```diff
@@ -132,11 +132,177 @@ kubectl create secret generic ceph-secret-admin
...
+
+The following steps have to be done at least once in order to be able to deploy CaaSP and CAP on top of OpenStack
+
+
+* Clone the terraform script
+
+  ```
+  git clone git@github.com:kubic-project/automation.git
+  cd automation/caasp-openstack-terraform
+  ```
```
Comment (cornelius) [inline Notes-CaaSP.md]: I would suggest to move this section to after creating the security groups. Then the terraform steps are all together.

File: Notes-CaaSP.md [hunk 4]
```diff
@@ -132,11 +132,177 @@ kubectl create secret generic ceph-secret-admin
...
+  terraform init
+  ```
+
+
+### Deploy CaaSP
+
+* Source the openrc.sh file, set the project and deploy
+
+  ```
+  . openrc.sh
```
Comment (cornelius) [inline Notes-CaaSP.md:243]: This already was done at the beginning after downloading the file.

File: Notes-CaaSP.md [hunk 5]
```diff
@@ -132,11 +132,177 @@ kubectl create secret generic ceph-secret-admin
...
+
+* Commands to run on the CaaSP nodes
+
+  This is only necessary if you use hostpath as storage class
+  ```
+  mkdir -p /tmp/hostpath_pv
+  chmod a+rwx /tmp/hostpath_pv
+  ```
+
+  Resize your root filesystem of the worker to match the disk provided by OpenStack (https://bugzilla.suse.com/show_bug.cgi?id=1069471#c12)
```
Comment (cornelius) [inline Notes-CaaSP.md]: I would remove the link to the Bugzilla entry because it's not accessible for everybody.

**Review by cornelius — CHANGES_REQUESTED** [review]
Instructions look good. I'm trying them right now. I commented on a few things I noticed when reviewing the text.

File: Notes-CaaSP.md (hunk 4 above)
Comment (greygoo) [reply Notes-CaaSP.md:243]: yes, but the steps above are only done once. Successive deployments will skip those and then it is not sourced.

**Review by cornelius — APPROVED** [review]

**Pull request closed without merging**